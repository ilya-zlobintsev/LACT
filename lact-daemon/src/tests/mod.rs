use insta::assert_json_snapshot;

use crate::{config::Config, server::handler::Handler};
use std::{fs, path::PathBuf};

#[tokio::test]
async fn snapshot_everything() {
    let test_data_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src/tests/data");

    for vendor_dir in fs::read_dir(test_data_dir).unwrap().flatten() {
        for device_dir in fs::read_dir(vendor_dir.path()).unwrap().flatten() {
            let test_key = format!(
                "{}/{}",
                vendor_dir.file_name().to_string_lossy(),
                device_dir.file_name().to_string_lossy()
            );

            let handler = Handler::with_base_path(&device_dir.path(), Config::default(), true)
                .await
                .unwrap();
            let mut device_info = handler
                .generate_snapshot_device_info()
                .into_values()
                .next()
                .unwrap();
            // Remove vulkan information, as it can only be fetched from the current running system and affects snapshot output
            device_info
                .get_mut("info")
                .unwrap()
                .as_object_mut()
                .unwrap()
                .remove("vulkan_info");

            assert_json_snapshot!(test_key, device_info);
        }
    }
}
