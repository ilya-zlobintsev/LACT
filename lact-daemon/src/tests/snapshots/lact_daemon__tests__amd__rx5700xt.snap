---
source: lact-daemon/src/tests/mod.rs
expression: device_info
---
{
  "clocks_info": {
    "max_mclk": 875,
    "max_sclk": 1900,
    "max_voltage": 1100,
    "table": {
      "type": "amd",
      "value": {
        "data": {
          "current_mclk_range": {
            "max": 875,
            "min": null
          },
          "current_sclk_range": {
            "max": 1900,
            "min": 1351
          },
          "od_range": {
            "curve_sclk_points": [
              {
                "max": 2150,
                "min": 800
              },
              {
                "max": 2150,
                "min": 800
              },
              {
                "max": 2150,
                "min": 800
              }
            ],
            "curve_voltage_points": [
              {
                "max": 1200,
                "min": 750
              },
              {
                "max": 1200,
                "min": 750
              },
              {
                "max": 1200,
                "min": 750
              }
            ],
            "mclk": {
              "max": 950,
              "min": 625
            },
            "sclk": {
              "max": 2150,
              "min": 800
            },
            "sclk_offset": null,
            "voltage_offset": null
          },
          "sclk_offset": null,
          "vddc_curve": [
            {
              "clockspeed": 1351,
              "voltage": 750
            },
            {
              "clockspeed": 1424,
              "voltage": 805
            },
            {
              "clockspeed": 1900,
              "voltage": 1100
            }
          ],
          "voltage_offset": null
        },
        "kind": "rdna"
      }
    }
  },
  "info": {
    "driver": "amdgpu",
    "drm_info": {
      "vram_clock_ratio": 1.0
    },
    "link_info": {
      "current_speed": "16.0 GT/s PCIe",
      "current_width": "16"
    },
    "pci_info": {
      "device_pci_info": {
        "model": "Navi 10 [Radeon RX 5600 OEM/5600 XT / 5700/5700 XT]",
        "model_id": "731F",
        "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
        "vendor_id": "1002"
      },
      "subsystem_pci_info": {
        "model": "Sapphire Technology Limited Navi 10 [Radeon RX 5600 OEM/5600 XT / 5700/5700 XT]",
        "model_id": "E409",
        "vendor": "Sapphire Technology Limited",
        "vendor_id": "1DA2"
      }
    },
    "vbios_version": "113-D1990103-O09",
    "vulkan_instances": []
  },
  "pci_info": {
    "device_pci_info": {
      "model": "Navi 10 [Radeon RX 5600 OEM/5600 XT / 5700/5700 XT]",
      "model_id": "731F",
      "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
      "vendor_id": "1002"
    },
    "subsystem_pci_info": {
      "model": "Sapphire Technology Limited Navi 10 [Radeon RX 5600 OEM/5600 XT / 5700/5700 XT]",
      "model_id": "E409",
      "vendor": "Sapphire Technology Limited",
      "vendor_id": "1DA2"
    }
  },
  "power_profile_modes": {
    "active": 6,
    "modes": {
      "0": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              5,
              1,
              0,
              4,
              800,
              4587520,
              -65536,
              0
            ]
          },
          {
            "clock_type": "SOCCLK",
            "values": [
              0,
              5,
              1,
              0,
              3,
              800,
              1310720,
              -6553,
              0
            ]
          },
          {
            "clock_type": "MEMCLK",
            "values": [
              0,
              5,
              1,
              0,
              4,
              800,
              327680,
              -65536,
              0
            ]
          }
        ],
        "name": "BOOTUP_DEFAULT"
      },
      "1": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              5,
              1,
              0,
              4,
              650,
              3932160,
              -6553,
              -65536
            ]
          },
          {
            "clock_type": "SOCCLK",
            "values": [
              0,
              5,
              1,
              850,
              4,
              800,
              1310720,
              -6553,
              0
            ]
          },
          {
            "clock_type": "MEMCLK",
            "values": [
              0,
              5,
              4,
              850,
              4,
              800,
              327680,
              -65536,
              0
            ]
          }
        ],
        "name": "3D_FULL_SCREEN"
      },
      "2": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              5,
              1,
              0,
              3,
              0,
              5898240,
              -65536,
              0
            ]
          },
          {
            "clock_type": "SOCCLK",
            "values": [
              0,
              5,
              1,
              0,
              3,
              0,
              1310720,
              -6553,
              0
            ]
          },
          {
            "clock_type": "MEMCLK",
            "values": [
              0,
              5,
              1,
              0,
              3,
              0,
              1966080,
              -65536,
              0
            ]
          }
        ],
        "name": "POWER_SAVING"
      },
      "3": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              5,
              1,
              0,
              4,
              500,
              4587520,
              -65536,
              0
            ]
          },
          {
            "clock_type": "SOCCLK",
            "values": [
              0,
              5,
              1,
              0,
              4,
              500,
              1310720,
              -6553,
              0
            ]
          },
          {
            "clock_type": "MEMCLK",
            "values": [
              0,
              5,
              1,
              0,
              4,
              500,
              1966080,
              -65536,
              0
            ]
          }
        ],
        "name": "VIDEO"
      },
      "4": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              5,
              4,
              1000,
              4,
              800,
              4587520,
              -65536,
              0
            ]
          },
          {
            "clock_type": "SOCCLK",
            "values": [
              0,
              5,
              1,
              0,
              4,
              800,
              327680,
              -65536,
              0
            ]
          },
          {
            "clock_type": "MEMCLK",
            "values": [
              0,
              5,
              1,
              0,
              4,
              800,
              327680,
              -65536,
              0
            ]
          }
        ],
        "name": "VR"
      },
      "5": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              5,
              4,
              1000,
              3,
              0,
              3932160,
              -65536,
              -65536
            ]
          },
          {
            "clock_type": "SOCCLK",
            "values": [
              0,
              5,
              4,
              850,
              3,
              0,
              327680,
              -65536,
              -32768
            ]
          },
          {
            "clock_type": "MEMCLK",
            "values": [
              0,
              5,
              4,
              850,
              3,
              0,
              327680,
              -65536,
              -32768
            ]
          }
        ],
        "name": "COMPUTE"
      },
      "6": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              5,
              4,
              1800,
              4,
              800,
              3932160,
              -65536,
              0
            ]
          },
          {
            "clock_type": "SOCCLK",
            "values": [
              0,
              5,
              1,
              850,
              4,
              800,
              1310720,
              -6553,
              0
            ]
          },
          {
            "clock_type": "MEMCLK",
            "values": [
              0,
              5,
              4,
              850,
              4,
              800,
              327680,
              -65536,
              0
            ]
          }
        ],
        "name": "CUSTOM"
      }
    },
    "value_names": [
      "FPS",
      "MinFreqType",
      "MinActiveFreqType",
      "MinActiveFreq",
      "BoosterFreqType",
      "BoosterFreq",
      "PD_Data_limit_c",
      "PD_Data_error_coeff",
      "PD_Data_error_rate_coeff"
    ]
  },
  "power_states": {
    "core": [
      {
        "enabled": true,
        "index": 0,
        "value": 300
      },
      {
        "enabled": true,
        "index": 1,
        "value": 1900
      }
    ],
    "vram": [
      {
        "enabled": true,
        "index": 0,
        "value": 100
      },
      {
        "enabled": true,
        "index": 1,
        "value": 500
      },
      {
        "enabled": true,
        "index": 2,
        "value": 625
      },
      {
        "enabled": true,
        "index": 3,
        "value": 875
      }
    ]
  },
  "stats": {
    "busy_percent": 0,
    "clockspeed": {
      "gpu_clockspeed": 10,
      "vram_clockspeed": 875
    },
    "fan": {
      "control_enabled": false,
      "pmfw_info": {},
      "pwm_current": 65,
      "pwm_max": 255,
      "pwm_min": 0,
      "speed_current": 29,
      "speed_max": 3400,
      "speed_min": 0
    },
    "memory_power_state": 3,
    "pcie_power_state": 1,
    "power": {
      "average": 29.0,
      "cap_current": 220.0,
      "cap_default": 220.0,
      "cap_max": 330.0,
      "cap_min": 110.0
    },
    "temps": {
      "edge": {
        "crit": 100.0,
        "crit_hyst": -273.1499938964844,
        "current": 58.0
      },
      "junction": {
        "crit": 108.0,
        "crit_hyst": -273.1499938964844,
        "current": 58.0
      },
      "mem": {
        "crit": 105.0,
        "crit_hyst": -273.1499938964844,
        "current": 62.0
      }
    },
    "voltage": {
      "gpu": 1100
    },
    "vram": {}
  }
}
